### PR TITLE
Travis -> GH Actions

### LINKED ISSUE #174: travis => GH actions
Upgrade to GitHub actions.

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by MetRonnie**
These changes close #174

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests.
- [x] No change log entry required (invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Comment by kinow** [comment]
@MetRonnie the function of a class changed. It had two string parameters. They were swapped in the code, but the tests were not updated. That's causing the test failures. Maybe we need something like `(bla, *, param1, static=...)` to force named parameters, to avoid this issue in the future.

The tests are passing for me with this patch

```diff
diff --git a/cylc/uiserver/tests/test_uiserver.py b/cylc/uiserver/tests/test_uiserver.py
index e3cfb6a..8a76a30 100644
--- a/cylc/uiserver/tests/test_uiserver.py
+++ b/cylc/uiserver/tests/test_uiserver.py
@@ -28,24 +28,24 @@ def test_my_application():
 
 def test_cylcuiserver_absolute_path():
     """Test a Cylc UI server created with absolute path for static assets."""
-    cylc_uiserver = CylcUIServer(8000, '/static/path', '/users/test')
+    cylc_uiserver = CylcUIServer(8000, '/users/test', static='/static/path')
     assert cylc_uiserver._port == 8000
-    assert cylc_uiserver._static == '/static/path'
+    assert cylc_uiserver.ui_path == '/static/path'
     assert cylc_uiserver._jupyter_hub_service_prefix == '/users/test'
 
 
 def test_cylcuiserver_relative_path():
     """Test a Cylc UI server created with relative path for static assets."""
-    cylc_uiserver = CylcUIServer(8000, './', '/users/test')
+    cylc_uiserver = CylcUIServer(8000, '/users/test', static='./')
     assert cylc_uiserver._port == 8000
-    print(cylc_uiserver._static)
+    print(cylc_uiserver.ui_path)
     # the code is using the directory relative to the main script ATM
-    assert cylc_uiserver._static.endswith('cylc/uiserver')
+    assert cylc_uiserver.ui_path.endswith('cylc/uiserver')
     assert cylc_uiserver._jupyter_hub_service_prefix == '/users/test'
 
 
 def test_cylcuiserver_create_static_handler():
-    cylc_uiserver = CylcUIServer(8000, './static', '/prefix/')
+    cylc_uiserver = CylcUIServer(8000, '/prefix/', static='./static')
     handler = cylc_uiserver._create_static_handler('(imgs/*.png)')
     assert "/prefix/((imgs/*.png))" == handler[0]
     assert issubclass(handler[1], StaticFileHandler)
@@ -53,7 +53,7 @@ def test_cylcuiserver_create_static_handler():
 
 
 def test_cylcuiserver_create_handler():
-    cylc_uiserver = CylcUIServer(8000, './', '/prefix/')
+    cylc_uiserver = CylcUIServer(8000, '/prefix/', static='./')
     handler = cylc_uiserver._create_handler('tests', RequestHandler,
                                             schema=True, table=False,
                                             number=1)
@@ -65,7 +65,7 @@ def test_cylcuiserver_create_handler():
 
 
 def test_cylcuiserver_create_graphql_handler():
-    cylc_uiserver = CylcUIServer(8000, './', '/prefix/')
+    cylc_uiserver = CylcUIServer(8000, '/prefix/', static='./')
     handler = cylc_uiserver._create_graphql_handler(
         'tests', TornadoGraphQLHandler, graphiql=True)
     assert "/prefix/tests" == handler[0]

```

**Review by kinow — COMMENTED** [review]
@MetRonnie the function of a class changed. It had two string parameters. They were swapped in the code, but the tests were not updated. That's causing the test failures. Maybe we need something like `(bla, *, param1, static=...)` to force named parameters, to avoid this issue in the future.

The tests are passing for me with this patch

```diff
diff --git a/cylc/uiserver/tests/test_uiserver.py b/cylc/uiserver/tests/test_uiserver.py
index e3cfb6a..8a76a30 100644
--- a/cylc/uiserver/tests/test_uiserver.py
+++ b/cylc/uiserver/tests/test_uiserver.py
@@ -28,24 +28,24 @@ def test_my_application():
 
 def test_cylcuiserver_absolute_path():
     """Test a Cylc UI server created with absolute path for static assets."""
-    cylc_uiserver = CylcUIServer(8000, '/static/path', '/users/test')
+    cylc_uiserver = CylcUIServer(8000, '/users/test', static='/static/path')
     assert cylc_uiserver._port == 8000
-    assert cylc_uiserver._static == '/static/path'
+    assert cylc_uiserver.ui_path == '/static/path'
     assert cylc_uiserver._jupyter_hub_service_prefix == '/users/test'
 
 
 def test_cylcuiserver_relative_path():
     """Test a Cylc UI server created with relative path for static assets."""
-    cylc_uiserver = CylcUIServer(8000, './', '/users/test')
+    cylc_uiserver = CylcUIServer(8000, '/users/test', static='./')
     assert cylc_uiserver._port == 8000
-    print(cylc_uiserver._static)
+    print(cylc_uiserver.ui_path)
     # the code is using the directory relative to the main script ATM
-    assert cylc_uiserver._static.endswith('cylc/uiserver')
+    assert cylc_uiserver.ui_path.endswith('cylc/uiserver')
     assert cylc_uiserver._jupyter_hub_service_prefix == '/users/test'
 
 
 def test_cylcuiserver_create_static_handler():
-    cylc_uiserver = CylcUIServer(8000, './static', '/prefix/')
+    cylc_uiserver = CylcUIServer(8000, '/prefix/', static='./static')
     handler = cylc_uiserver._create_static_handler('(imgs/*.png)')
     assert "/prefix/((imgs/*.png))" == handler[0]
     assert issubclass(handler[1], StaticFileHandler)
@@ -53,7 +53,7 @@ def test_cylcuiserver_create_static_handler():
 
 
 def test_cylcuiserver_create_handler():
-    cylc_uiserver = CylcUIServer(8000, './', '/prefix/')
+    cylc_uiserver = CylcUIServer(8000, '/prefix/', static='./')
     handler = cylc_uiserver._create_handler('tests', RequestHandler,
                                             schema=True, table=False,
                                             number=1)
@@ -65,7 +65,7 @@ def test_cylcuiserver_create_handler():
 
 
 def test_cylcuiserver_create_graphql_handler():
-    cylc_uiserver = CylcUIServer(8000, './', '/prefix/')
+    cylc_uiserver = CylcUIServer(8000, '/prefix/', static='./')
     handler = cylc_uiserver._create_graphql_handler(
         'tests', TornadoGraphQLHandler, graphiql=True)
     assert "/prefix/tests" == handler[0]

```

File: cylc/uiserver/tests/test_workflows_mgr.py [hunk 1]
```diff
@@ -376,32 +380,6 @@ async def test_unregister(
     assert workflow_name not in uiserver.workflows_mgr.inactive
 
 
-@pytest.mark.asyncio
-async def test_unregister(
```
Comment (MetRonnie) [inline cylc/uiserver/tests/test_workflows_mgr.py:380]: Duplicate of test above

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: Test
+
+on:
+  pull_request:
+  workflow_dispatch:
+  push:
+    branches: [master]
```
Comment (MetRonnie) [inline .github/workflows/test.yml:7]: Useful to use `workflow_dispatch` so we can manually trigger the workflow from the repo's Actions tab if needed

File: cylc/uiserver/tests/test_uiserver.py [hunk 3]
```diff
@@ -28,32 +28,32 @@ def test_my_application():
 
 def test_cylcuiserver_absolute_path():
     """Test a Cylc UI server created with absolute path for static assets."""
-    cylc_uiserver = CylcUIServer(8000, '/static/path', '/users/test')
+    cylc_uiserver = CylcUIServer(8000, '/users/test', static='/static/path')
     assert cylc_uiserver._port == 8000
-    assert cylc_uiserver._static == '/static/path'
+    assert cylc_uiserver.ui_path == '/static/path'
```
Comment (MetRonnie) [inline cylc/uiserver/tests/test_uiserver.py:33]: Thanks for the fixes, Bruno!

Comment (kinow) [reply cylc/uiserver/tests/test_uiserver.py:33]: Thank **you** for fixing CI for the UIS Ronnie!

**Review by kinow — APPROVED** [review]

**Review by datamel — APPROVED** [review]
Had to do a bit of reading around as still a bit of a github actions workflow novice, hope I haven't missed anything but it all looks pretty sound to me. Thanks Ronnie!

File: .github/workflows/test.yml (hunk 2 above)
Comment (MetRonnie) [reply .github/workflows/test.yml:7]: Another possibly useful thing to note @datamel, the `on: push: branches: [master]` happens when you merge PRs too, e.g. when this PR itself was just merged https://github.com/cylc/cylc-uiserver/actions/runs/633233211

**Comment by oliver-sanders** [comment]
Thanks for the MacOS build, really helpful.

**Pull request merged**